processor/src/test/java/org/mapstruct/ap/test/selection/qualifier/iterable/TopologyMapper.java

```diff
@@ -0,0 +1,78 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.selection.qualifier.iterable;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Qualifier;
+import org.mapstruct.factory.Mappers;
+
+/**
+ *
+ * @author Sjaak Derksen
+ */
+@Mapper
+public abstract class TopologyMapper {
+
+    public static final TopologyMapper INSTANCE = Mappers.getMapper( TopologyMapper.class );
+
+    @Mapping( target = "topologyFeatures", qualifiedBy = Rivers.class )
+    public abstract TopologyEntity mapTopology(TopologyDto dto);
```

Comment:
> iterable and that a new forged iterable method is created for that one instead?

Hmmm.. Just did that.. Doesn't work... 😢  Need to think about what makes an `IterableMethod` equal